File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/filter/AccessLogFilter.java
package com.jmsoftware.maf.springcloudstarter.filter;

import com.jmsoftware.maf.springcloudstarter.configuration.MafConfiguration;
import com.jmsoftware.maf.springcloudstarter.util.RequestUtil;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import org.springframework.util.AntPathMatcher;
import org.springframework.web.filter.OncePerRequestFilter;

import javax.servlet.FilterChain;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * <h1>RequestFilter</h1>
 * <p>Request filter.</p>
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com
 * @date 2019-03-23 14:24
 **/
@Slf4j
@Component
@RequiredArgsConstructor
public class AccessLogFilter extends OncePerRequestFilter {
    private final MafConfiguration mafConfiguration;
    private final AntPathMatcher antPathMatcher = new AntPathMatcher();

    @Override
    @SuppressWarnings("NullableProblems")
    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
                                    FilterChain filterChain) throws IOException, ServletException {
        // Ignore URL
        for (String ignoredUrl : this.mafConfiguration.flattenIgnoredUrls()) {
            if (this.antPathMatcher.match(ignoredUrl, request.getRequestURI())) {
                filterChain.doFilter(request, response);
                return;
            }
        }
        log.info("The requester({}) requested resource. Request URL: [{}] {}", RequestUtil.getRequestIpAndPort(request),
                 request.getMethod(), request.getRequestURL());
        filterChain.doFilter(request, response);
    }
}

File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/filter/AccessLogFilter.kt
package com.jmsoftware.maf.springcloudstarter.filter

import com.jmsoftware.maf.common.util.logger
import com.jmsoftware.maf.springcloudstarter.property.MafConfigurationProperties
import com.jmsoftware.maf.springcloudstarter.util.RequestUtil
import org.springframework.stereotype.Component
import org.springframework.util.AntPathMatcher
import org.springframework.web.filter.OncePerRequestFilter
import javax.servlet.FilterChain
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * # RequestFilter
 *
 * Request filter.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/12/22 9:55 PM
 */
@Component
class AccessLogFilter(
    private val mafConfigurationProperties: MafConfigurationProperties,
) : OncePerRequestFilter() {
    companion object {
        private val log = logger()
    }

    private val antPathMatcher = AntPathMatcher()

    override fun doFilterInternal(
        request: HttpServletRequest,
        response: HttpServletResponse,
        filterChain: FilterChain
    ) {
        // Ignore URL
        for (ignoredUrl in mafConfigurationProperties.flattenIgnoredUrls()) {
            if (antPathMatcher.match(ignoredUrl, request.requestURI)) {
                filterChain.doFilter(request, response)
                return
            }
        }
        log.info("The requester(${RequestUtil.getRequestIpAndPort(request)}) requested resource. Request URL: [${request.method}] ${request.requestURL}")
        filterChain.doFilter(request, response)
    }
}
